core/chains/evm/gas/fee_history_estimator.go

```diff
@@ -0,0 +1,424 @@
+	if originalGasPrice == nil || originalGasPrice.Cmp(maxPrice) >= 0 {
+		return nil, 0, fmt.Errorf("%w: error while retrieving original gas price: originalGasPrice: %s. Maximum price configured: %s",
+			commonfee.ErrBump, originalGasPrice, maxPrice)
+	}
+
+	currentGasPrice, err := f.getGasPrice()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	bumpedGasPrice := originalGasPrice.AddPercentage(f.config.BumpPercent)
+	bumpedGasPrice, err = LimitBumpedFee(originalGasPrice, currentGasPrice, bumpedGasPrice, maxPrice)
+	if err != nil {
+		return nil, 0, fmt.Errorf("gas price error: %s", err.Error())
+	}
+
+	f.logger.Debugw("bumped gas price", "originalGasPrice", originalGasPrice, "bumpedGasPrice", bumpedGasPrice)
+
+	return bumpedGasPrice, gasLimit, nil
+}
+
+// BumpDynamicFee provides a bumped dynamic fee by bumping the previous one by BumpPercent.
+// If the original values are higher than the max price it returns an error as there is no room for bumping. If maxPriorityFeePerGas is bumped
+// above the priority fee threshold then there is a good chance there is a connectivity issue and we shouldn't bump.
+// Both maxFeePerGas as well as maxPriorityFeePerGas need to be bumped otherwise the RPC won't accept the transaction and throw an error.
+// See: https://github.com/ethereum/go-ethereum/issues/24284
+// It aggregates the market, bumped, and max price to provide a correct value, for both maxFeePerGas as well as maxPriorityFerPergas.
+func (f *FeeHistoryEstimator) BumpDynamicFee(ctx context.Context, originalFee DynamicFee, maxPrice *assets.Wei, _ []EvmPriorAttempt) (bumped DynamicFee, err error) {
+	// For chains that don't have a mempool there is no concept of gas bumping so we force-call FetchDynamicPrice to update the underlying base fee value
+	if f.config.BlockHistorySize == 0 {
+		if err = f.RefreshDynamicPrice(); err != nil {
+			return
+		}
+		return f.GetDynamicFee(ctx, maxPrice)
+	}
+	// Sanitize original fee input
+	// According to geth's spec we need to bump both maxFeePerGas and maxPriorityFeePerGas for the new attempt to be accepted by the RPC
+	if originalFee.FeeCap == nil ||
+		originalFee.TipCap == nil ||
+		((originalFee.TipCap.Cmp(originalFee.FeeCap)) > 0) ||
+		(originalFee.FeeCap.Cmp(maxPrice) >= 0) {
+		return bumped, fmt.Errorf("%w: error while retrieving original dynamic fees: (originalFeePerGas: %s - originalPriorityFeePerGas: %s). Maximum price configured: %s",
+			commonfee.ErrBump, originalFee.FeeCap, originalFee.TipCap, maxPrice)
+	}
+
+	currentDynamicPrice, err := f.getDynamicPrice()
+	if err != nil {
+		return
+	}
+
+	bumpedMaxPriorityFeePerGas := originalFee.TipCap.AddPercentage(f.config.BumpPercent)
+	bumpedMaxFeePerGas := originalFee.FeeCap.AddPercentage(f.config.BumpPercent)
+
+	bumpedMaxPriorityFeePerGas, err = LimitBumpedFee(originalFee.TipCap, currentDynamicPrice.TipCap, bumpedMaxPriorityFeePerGas, maxPrice)
+	if err != nil {
+		return bumped, fmt.Errorf("maxPriorityFeePerGas error: %s", err.Error())
```

Comment:
Same here and below for the limit bumped fee error
```suggestion
		return bumped, fmt.Errorf("failed to limit bumped fee: %s", err.Error())
```

Reply:
Left a comment above

Reply:
Same here. So something like `failed to bump max priority fee`. Then below we can do `failed to bump max fee`